### PR TITLE
fix: Moved expires_in field after scope in OAuth2 DatasourceForm #31059

### PR DESCRIPTION
**Pull request by theAravinthM**
## Moved the `expires_in` field after `scope` field while configuring OAuth2 datasource - for better user experience
PR for the issue: #31059 
## Description
__While configuring the OAuth2 Datasource, when the `Grant_type` field is selected as `Authorization Code`, the `authorization_expires_in` is placed right after the `scope` field.__
__This is to help users find all the related fields at one place. This improves User Experience.__
- `fixes #31059`
- Number of files modified: `1`
- Name of the file modified: `appsmith/app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx`

## Automation
/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!CAUTION]  
> If you modify the content in this section, you are likely to disrupt the CI result for your PR.

## Summary by CodeRabbit

- **New Features**
	- Enhanced OAuth2 authorization code flow in the DataSource Editor by introducing specific rendering improvements for a more intuitive user experience.

### CODE REVIEW

File: app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx [hunk 1]
```diff
@@ -800,17 +800,133 @@ class DatasourceRestAPIEditor extends React.Component<Props> {
...
+            "Client Authentication",
+            "",
+            false,
+            "",
+            !!_.get(formData.authentication, "isAuthorizationHeader"),
+          )}
+        </FormInputContainer>
+      </>
+    );
+  };
```
[the remark this answers is not in the document]
Comment (theAravinthM) [reply app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx:919]: Alright. I will do the refactoring!

**Comment by sneha122** [comment]
@theAravinthM Hi I tested the PR using this link https://ce-31863.dp.appsmith.com/ I don't see the expires_in field at all even when I select `Authorization code` as grant type. Please check

**Comment by theAravinthM** [comment]
@sneha122 Hi ! That was by mistake, since I could not test my code like the link that you provided. I have modified a little chunk of my code from the previous commits. 
Thanks for the review and now I hope my code would fit in!

**Comment by sneha122** [comment]
Hi @theAravinthM

On client side as well there is a linting failure in file src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx, can you please format this file using prettier formatter?

These should fix all of your linting errors and we can merge the PR as quickly as we can

**Comment by theAravinthM** [comment]
Hi @sneha122 

Thanks for the review! I have formatted the code with Prettier code formatter.

**Review by sneha122 — APPROVED** [review]

**Comment by theAravinthM** [comment]
Hi @sneha122 
Happy that many tests have been passed on this PR and there is one more test named `perform-test / ci-test-result` is in the Waiting status. I went through several online blogs which asks to make a blank commit. Just asking if I need to do that. 
Thanks!

**Comment by sneha122** [comment]
Hi @theAravinthM Thanks for checking back, but there is some issue in running automated tests on this PR, let me investigate this and get back to you

**Comment by sneha122** [comment]
@theAravinthM Can you please pull latest release branch in your forked repo and merge it into this branch?

**Comment by sneha122** [comment]
Screenshot 2024-04-08 at 2 10 54 PM
It looks like sync has not been successful, I still see forked repo behind the original one, please sync it first and then merge latest release into it, Thanks!

**Comment by theAravinthM** [comment]
Oh my bad! I just mistakenly synced the release branch of my forked repo before. 
Now I have synced this branch. Sorry for the inconvenience caused! Thanks!

**Comment by theAravinthM** [comment]
Hi @sneha122 I am really unsure why that `perform-test / ci-test-result` test is still in the waiting status. Is there a way that could be resolved? Thanks!

**Pull request closed without merging**